File: examples/examples_CPU_FORTRAN/gm29.f90

PROGRAM GM29_Example

   PARAMETER (N=100000000)
   INTEGER i,GM29_generate,sum
   REAL val,rsum

   TYPE GM29_state
     INTEGER z(68)
   END TYPE GM29_state
   TYPE(GM29_state) state

   CALL GM29_init ( state )
   CALL GM29_Print_State (state)

   sum = 0

   DO i=1,N
     sum = sum + GM29_generate (state)
   END DO

   WRITE(*,1) N
   rsum = sum/4294967296.
   val = GM29_generate (state)/4294967296.
   IF (rsum<0) THEN
     rsum=rsum+1
   ENDIF
   IF (val<0) THEN
     val=val+1
   ENDIF
   WRITE(*,2) rsum
   WRITE(*,3) val

1  FORMAT(I9," GM29 pseudorandom numbers generated.")
2  FORMAT("Fractional part of the total sum of generated numbers: ",F8.6)
3  FORMAT("Next output value: ",F8.6)

END PROGRAM GM29_Example

File: examples/examples_CPU_C/gm29.c

#include<stdio.h>
#include<gm29.h>

#define NN       100000000UL

int main(void){ 
   long i; unsigned int sum=0;
   gm29_state state;
   gm29_init_(&state);
   gm29_print_state_(&state);
   for(i=0;i<NN;i++) sum+=gm29_generate_(&state);
   printf("%ld GM29 pseudorandom numbers generated.\n",NN);
   printf("Fractional part of the total sum of generated numbers: %f\n",sum/4294967296.);
   printf("Next output value: %f\n",gm29_generate_(&state)/4294967296.);
   return 0;
}
